### PR TITLE
Document process for enabling Vertical Pod Autoscalers collector

### LINKED ISSUE #1408: Improve (!?) documentation for enabling VerticalPodAutoscaler
**What happened**:

It took me more time than I would have liked to understand how to enable `VerticalPodAutoscaler` metrics.

The documentation [hints](https://github.com/kubernetes/kube-state-metrics#enabling-verticalpodautoscalers) at the solution but would benefit from an additional explanation for running `kube-state-metrics` with `--resource=${RESOURCES},verticalpodautoscalers`.

**What you expected to happen**:

A more detailed explanation of what's needed and how to enable this with e.g. Prometheus Operator.

Perhaps back-filling issues with VPA and the Prometheus Operator to provide guidance on this too.

It's very helpful just confusing to resolve.

**How to reproduce it (as minimally and precisely as possible)**:

I'm running MicroK8s 1.20 and `microk8s enable prometheus` and have VPA installed:

```bash
kubectl get crd/verticalpodautoscalers.autoscaling.k8s.io \
--output=jsonpath="{.spec.versions[*].name}"
v1beta1 v1beta2 v1
```
After cloning this repo and deploying `./examples/standard`, I realized that the Prometheus Operator bundles `kube-state-metrics` but then realized that it does not enable `verticalpodautoscalers` (by default).

```bash
kubectl logs deployment/kube-state-metrics \
--namespace=monitoring \
--container=kube-state-metrics

...
Active resources:
  certificatesigningrequests,configmaps,cronjobs,
  daemonsets,deployments,endpoints,horizontalpodautoscalers,
  ingresses,jobs,leases,limitranges,mutatingwebhookconfigurations,
  namespaces,networkpolicies,nodes,persistentvolumeclaims,
  persistentvolumes,poddisruptionbudgets,pods,replicasets,
  replicationcontrollers,resourcequotas,secrets,services,
  statefulsets,storageclasses,validatingwebhookconfigurations,
  volumeattachments
...
```
and:

```bash
kubectl get deployment/kube-state-metrics \
--namespace=monitoring \
--output=jsonpath="{.spec.template.spec.containers[?(@.name=='kube-state-metrics')]}" \
| jq .
```

Yields:

```JSON
{
  "args": [
    "--host=127.0.0.1",
    "--port=8081",
    "--telemetry-host=127.0.0.1",
    "--telemetry-port=8082",
  ],
  "image": ".../kube-state-metrics:v1.9.7",
  "imagePullPolicy": "IfNotPresent",
  "name": "kube-state-metrics",
  "resources": {},
  "terminationMessagePath": "/dev/termination-log",
  "terminationMessagePolicy": "File"
}
```

I don't know how to configure the metrics correctlypermanently but I managed to tweak it:

```bash
RESOURCES="certificatesigningrequests,configmaps,cronjobs,"\
"daemonsets,deployments,endpoints,horizontalpodautoscalers,"\
"ingresses,jobs,leases,limitranges,mutatingwebhookconfigurations,"\
"namespaces,networkpolicies,nodes,persistentvolumeclaims,"\
"persistentvolumes,poddisruptionbudgets,pods,replicasets,"\
"replicationcontrollers,resourcequotas,secrets,services,"\
"statefulsets,storageclasses,validatingwebhookconfigurations,"\
"verticalpodautoscalers,volumeattachments"

PATCH="[{"\
"'op': 'add',"\
"'path':'/spec/template/spec/containers/0/args/-',"\
"'value':\"--resources=${RESOURCES}\"}]"


kubectl patch deployment/kube-state-metrics \
--namespace=monitoring \
--type=json \
--patch="${PATCH}" 
```

**Anything else we need to know?**:

Hopefully others who struggle with this will find this issue.

I blogged about the experience [here](https://pretired.dazwilkin.com/posts/210305/)

**Environment**:
- kube-state-metrics version: 
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Other info:

### PR DESCRIPTION
**Pull request by DazWilkin**
**What this PR does / why we need it**:

More fully document enabling Vertical Pod Autoscalers collector.

**Which issue(s) this PR fixes**:

Fixes #1408

### CODE REVIEW

**Comment by DazWilkin** [comment]
CLA signed

File: README.md [hunk 1]
```diff
@@ -120,9 +120,10 @@ that ensures that there are no possible conflicts.
 
 #### Enabling VerticalPodAutoscalers
 
-Please note that the collector for `verticalpodautoscalers` are disabled by default.
-This is because Vertical Pod Autoscalers are managed as custom resources. If you want to enable this collector,
-please ensure that you have the `v1beta2` CRDs installed beforehand. They can be found [here](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/deploy/vpa-beta2-crd.yaml).
+Please note that the collector for `verticalpodautoscalers` is **disabled** by default; Vertical Pod Autoscaler metrics will not be collected until the collector is enabled. This is because Vertical Pod Autoscalers are managed as custom resources.
+
+If you want to enable this collector,
+please review the [instructions](./docs/verticalpodautoscaler-metrics.md#Configuration) in the [Vertical Pod Autoscaler Metrics](./docs/verticalpodautoscaler-metrics.md) documentation.
```
Comment (lilic) [inline README.md:126]:
```suggestion
the [instructions](./docs/verticalpodautoscaler-metrics.md#Configuration) are located in the [Vertical Pod Autoscaler Metrics](./docs/verticalpodautoscaler-metrics.md) documentation.
```

File: docs/verticalpodautoscaler-metrics.md [hunk 2]
```diff
@@ -11,6 +11,97 @@
...
+Vertical Pod Autoscalers(VPAs) are managed as custom resources.
+
+To enable the Vertical Pod Autoscaler collector, please:
+
+1. Ensure that the Vertical Pod Autoscaler CRDs are installed in the cluster. The CRDs are [here](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/deploy/vpa-beta2-crd.yaml).
+2. Ensure that `verticalpodautoscalers` is included in list of `Resources` enabled using the flag `--resources` when `kube-state-metrics` is run (see below).
+
+One of the [command line arguments](./docs/cli-arguments.md) for `kube-state-metrics` is `--resources`. If this flag is omitted, a default set of Resources is enabled. This default list does **not** include Vertical Pod Autoscalers.
+
+To enable Vertical Pod Autoscalers, the `kube-state-metrics` flag `--resource` must be included when the binary is run and the list of resources must include `verticalpodautoscalers`.
```
Comment (lilic) [inline docs/verticalpodautoscaler-metrics.md:25]:
```suggestion
To enable Vertical Pod Autoscalers, the `kube-state-metrics` flag `--resource` must be included when the binary is run and the list of resources must include `verticalpodautoscalers`, e.g. `kube-state-metrics --resources=verticalpodautoscalers`.
```

**Review by lilic — COMMENTED** [review]
Few nits really, otherwise looks good to me, thanks for contributing this!

File: docs/verticalpodautoscaler-metrics.md (hunk 2 above)
Comment (DazWilkin) [reply docs/verticalpodautoscaler-metrics.md:25]: Generally, developers will need to include a long list of values for the `--resources` flag including `verticalpodautoscalers`. `--resources=verticalpodautoscalers` on its own would *only* enable `verticalpodautoscalers` and disable all the default resources.

**Review by lilic — APPROVED** [review]
Nice, thank you! 🎉 
/lgtm

**Pull request merged**